src/AccessibilityInsights.SharedUx/Controls/CustomControls/ScannerResultCustomListContext.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Windows;
+using System.Windows.Controls;
+
+namespace AccessibilityInsights.SharedUx.Controls.CustomControls
+{
+    class ScannerResultCustomListContext
+    {
+        internal Action UpdateTree { get; }
+        internal Action SwitchToServerLogin { get; }
+        internal bool ShowAllResults;
+        internal Button BtnShowAll;
+        internal FrameworkElement DataContext;
+        internal Guid EcId;
+
+        public ScannerResultCustomListContext( Action updateTree, Action switchToServerLogin, Button btnShowAll, FrameworkElement dataContext, Guid ecId)
+        {
+            UpdateTree = updateTree ?? throw new ArgumentNullException(nameof(updateTree));
+            SwitchToServerLogin = switchToServerLogin ?? throw new ArgumentNullException(nameof(switchToServerLogin));
+            BtnShowAll = btnShowAll ?? throw new ArgumentNullException(nameof(btnShowAll));
+            DataContext = dataContext ?? throw new ArgumentNullException(nameof(dataContext));
+            EcId = ecId;
+        }
+
+        public void changeVisibility()
```

Comment:
If I'm reading this correctly, this method could be completely inside the parent. If that's correct, I'd recommend adding `Action ChangeAction { get; }` as a context property. Then have a method in `ScannerResultControl`:

```
private void ChangeVisibility()
{
    var visible = this.btnShowAll.Visibility;
    this.ShowAllResults = visibile == Visibility.Collapsed;
    UpdateTree();
    this.btnShowAll.Visibility = visible;
}
```